### ISSUE TITLE
Call multiple tasks from CLI and pass them to RunTarget

### ISSUE THREAD

**Issue by HofmeisterAn** [cake-build/cake#2470]
I used a lot of Gradle in the past. One of the most features I miss in Cake, is to run multiple tasks from CLI. In Gradle I could simple run `.\gradlew clean build`. In Cake I’m forced to use `IsDependentOn`, which does not make sense in every use case (I don’t want to run clean on every build).

It bothers me especial on my build environment (CI). It would be a lot easier to call multiple tasks in one stage. I think this would be a great feature and make Cake much better.

1. Pass a list of arguments to `RunTarget`.
2. Use the argument order and call every task including `IsDependentOn` tasks.
3. Consider the tree dependency of the tasks, to order them as well.

**Comment by cbittencourt** [comment]
I am also interested in this feature. We have modified build.cake to call RunTarget(target) within a `foreach`, but I've noticed that we can run two tasks that are dependent on the same data from `Setup<T>()` . 

```
Setup<MyData>(setupContext => {
  return new MyData()
    {
      DataItem = Argument<string>("command-arg","")
    };
});

Task("Task1").Does<MyData>((myData) => { 
  Information("Task1 runs successfully."); 
});

Task("Task2").Does<MyData>((myData) => { 
  Error("Task2 fails with `Context data of type 'Submission#0+MyData' has already been registered.`"); 
});

foreach(var target in targets) {
	RunTarget(target);
}
```

Command for example above:
`./build.ps1 -targets Task1,Task2 --command-arg=foo`

**Comment by Mrks83** [comment]
Coming from MSBuild I would appreciate this feature, too.

**Comment by FrankRay78** [comment]
I'd like to have a go at this one @augustoproiete - please can you assign it to me?

**Comment by devlead** [comment]
I would expect this to not rely on string splitting but on the fact that Cake allows specifying the same argument multiple times, i.e.
```PowerShell
dotnet cake runtargets.cake --target=A --target=B --target=C
```
and a cake script could look something like this
```cake
var targets = Arguments<string>("target", "A");

Task("A");

Task("B");

Task("C");

RunTargets(targets);
```
a way to simulate this now would be to add a meta task in a Cake script i.e.
```cake
CakeReport RunTargets(ICollection<string> targets)
    => RunTarget(GetOrAddTargetsTask(targets).Name);

Task<CakeReport> RunTargetsAsync(ICollection<string> targets)
    => RunTargetAsync(GetOrAddTargetsTask(targets).Name);

private ICakeTaskInfo GetOrAddTargetsTask(ICollection<string> targets)
{
    var targetsTaskName = string.Join(',', targets);
    var targetsTask = Tasks.FirstOrDefault(
        task => task.Name == targetsTaskName
    );
    if (targetsTask == null)
    {
        var task = Task(targetsTaskName);
        foreach(var target in targets)
        {
            task.IsDependentOn(target);
        }
        targetsTask = task.Task;
    }
    return targetsTask;
}
```
as PoC gist here: https://gist.github.com/devlead/84b21afe5688e638a86f85b940ae57be

For that i.e.
```PowerShell
dotnet cake runtargets.cake --target=A
```
will result in
```
========================================
A
========================================

Task                          Duration            
--------------------------------------------------
--------------------------------------------------
Total:                        00:00:00.0202365    
```
and 
```PowerShell
dotnet cake runtargets.cake --target=A --target=B
```
will result in
```

========================================
A
========================================

========================================
B
========================================

========================================
A,B
========================================

Task                          Duration            
--------------------------------------------------
--------------------------------------------------
Total:                        00:00:00.0251612    
```
and
```PowerShell
dotnet cake runtargets.cake --target=A --target=B --target=C
```
will result in
```

========================================
A
========================================

========================================
B
========================================

========================================
C
========================================

========================================
A,B,C
========================================

Task                          Duration            
--------------------------------------------------
--------------------------------------------------
Total:                        00:00:00.0560336    
```

**Comment by FrankRay78** [comment]
Thanks @devlead, I'll put what you suggest into a PR for review...

**Comment by FrankRay78** [comment]
I have now raised a PR for this issue, see: https://github.com/cake-build/cake/pull/4054

**Comment by devlead** [comment]
Fixed by #4054
